pull/context.go

```diff
@@ -200,3 +204,67 @@ type Reviewer struct {
 	Name    string
 	Removed bool
 }
+
+type Signature interface {
+	GetEmail() string
+	GetIsValid() bool
+	GetPayload() string
+	GetSignature() string
+	GetSigner() string
+	GetState() string
+	GetSignedByGitHub() bool
+}
```

Comment:
I'm not sold on the use of an interface to represent the different signature types here, especially since the only difference is the addition of the ID field in the GPG case. It stands out in comparison to the rest of the `pull.Context` return types, which are all simple concrete structs.

Using a pointer to a concrete `Signature` in `Commit` also makes it clearer that the value may not be present.

What do you think about using a concrete type with an optional `KeyID` field and maybe a `Type` field (similar to the `Type` field on `Reviewer`, especially since you already defined a type type for use with GraphQL)? Since you already have functions to convert the GitHub API types into the `pull.Context` types, I don't think it would change the code much.

Reply:
That's fair. I mostly went with the interface out of habit rather than a particular need.

You're right it didn't change the code much and arguably given then types involved it feels neater now.